frontend/routes/notifications/notifications.component.tsx

```diff
@@ -0,0 +1,179 @@
+import { NotificationEmptyItem } from "./notifications.emptyItem";
+import { NotificationsPanel } from "./notifications.panel";
+import { simpleDate, getSunday } from "../../components/utils/js/utils.filter";
+import { BarIconButton, UserActionButton, ItemLabel } from "../components/components.styles";
+import {toArray, groupBy, sortBy } from "lodash";
+import { NotificationsPanelItem, NotificationsPanelHeader } from "./notifications.panel.styles";
+
+// Props bound in <file://./notifications.container.ts>
+interface IProps {
+	sendGetNotifications: () => void ; // Bound to redux action sendGetNotifications
+	confirmSendDeleteAllNotifications: () => void ; // Bound to redux saga action confirmSendDeleteAllNotifications
+	sendUpdateNotificationRead: (id: string, read: boolean) => void; // Bound to redux saga sendUpdateNotificationRead
+	sendDeleteNotification: (id: string) => void; // Bound to redux saga sendDeleteNotification
+	notifications: INotification[]; // Bound to store state notifications
+}
+
+const NotificationsDate = () =>
+	(<NotificationsPanelItem>
+			<NotificationsPanelHeader>
+				<ItemLabel>
+					{simpleDate(new Date())}
+				</ItemLabel>
+				<ItemLabel/>
+			</NotificationsPanelHeader>
+	</NotificationsPanelItem>);
+
+export class Notifications extends React.PureComponent<IProps, any> {
+	public state = {
+		open: false,
+		menuElement: null
+	};
+
+	public componentDidMount = () => {
+		// This will download notifications from the server and save to the store on init
+		this.props.sendGetNotifications();
+	}
+
+	public toggleDrawer = (e: React.SyntheticEvent) => {
+		this.setState(Object.assign({open: !this.state.open }));
+	}
+
+	public toggleMenu = (e: React.SyntheticEvent) => {
+		this.setState({ menuElement: this.state.menuElement ? null : e.currentTarget });
+		return false;
+	}
+
+	public deleteAllNotifications = (e: React.SyntheticEvent) => {
+		this.toggleMenu(e);
+		this.props.confirmSendDeleteAllNotifications();
+	}
+
+	public thisWeeksNotifications = (notifications) => {
+		const lastSunday = getSunday().getTime();
+		return notifications.filter((n) => n.timestamp > lastSunday);
+	}
+
+	public lastWeeksNotifications = (notifications) => {
+		const lastSunday = getSunday().getTime();
+		const prevSunday = getSunday(-1).getTime();
+		return notifications.filter((n) => n.timestamp > prevSunday && n.timestamp < lastSunday );
+	}
+
+	public moreThanTwoWeeksAgoNotifications = (notifications) => {
+		const prevSunday = getSunday(-1).getTime();
+		return notifications.filter((n) => n.timestamp < prevSunday );
+	}
+
+	public hasNotifications = () => {
+		return this.props.notifications.length > 0;
+	}
+
+	public renderNotificationsHeader = () => {
+		return (<ListSubheaderToolbar rightContent={
+					<>
+					<BarIconButton aria-label="Menu" style={{marginRight: -18}} onClick={this.toggleMenu}>
+						<Icon>more_vert</Icon>
+						<Menu
+							anchorEl={this.state.menuElement}
+							open={!!this.state.menuElement}
```

Comment:
```suggestion
							open={Boolean(this.state.menuElement)}
```

Reply:
I prefer the other style , is more concise and style readable.